src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -755,10 +966,11 @@ static int hwmon_sdm_update_sensors_by_type(struct platform_device *pdev,
 	if (!sdm->privileged)
 		return hwmon_sdm_read_from_peer(pdev, repo_type);
 
-	sdm->sensor_data[repo_id] = (char*)kzalloc(sizeof(char) * resp_size,
-                                               GFP_KERNEL);
+	if (!sdm->sensor_data[repo_id])
+		sdm->sensor_data[repo_id] = (char*)kzalloc(sizeof(char) * RESP_LEN, GFP_KERNEL);
```

Comment:
Can we avoid this memory allocation as well? Where is this memory freed? In general, the less memory allocation, the better in driver... memory allocation is slow and difficult to manage.

Reply:
Hi Max, sensor_data size is known at runtime by issuing sensor size request before the actual request. Since, this work is pending, we are allocating 4KB memory by default for now. 
And I agree that we need to free the memory in device_remove () function. But, I replaced kzalloc() with devm_kzalloc() API now, so it is not required to free the memory.  Because, when the device is detached or the driver for the device is unloaded, that memory is freed automatically.